File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/MapRunningFightListMessage.scala





















// Generated on 05/03/2013 10:34:42
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait MapRunningFightListMessage extends MessageDeserializer {
  val messageId = 5743

  protected trait Message extends BaseMessage with Serializable {
    val messageId = MapRunningFightListMessage.messageId

    val fights: Seq[FightExternalInformations.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeUShort(fights.length)
      for (entry <- fights) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(fights: Seq[FightExternalInformations.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val fights = {
      val builder = collection.immutable.Seq.newBuilder[FightExternalInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += FightExternalInformations.deserialize(buf)
      }
      builder.result()
    }

    create(fights)
  }

}

object MapRunningFightListMessage extends MapRunningFightListMessage {
  case class MessageImpl(fights: Seq[FightExternalInformations.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(fights: Seq[FightExternalInformations.TypeImpl]) = MessageImpl(fights)

  def apply(fights: Seq[FightExternalInformations.TypeImpl]) = create(fights)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/MapRunningFightListMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MapRunningFightListMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5743;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public FightExternalInformations[] fights;
    
    public MapRunningFightListMessage() { }
    
    public MapRunningFightListMessage(FightExternalInformations[] fights) {
        this.fights = fights;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(fights.length);
        for (FightExternalInformations entry : fights) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        fights = new FightExternalInformations[limit];
        for (int i = 0; i < limit; i++) {
            fights[i] = new FightExternalInformations();
            fights[i].deserialize(buf);
        }
    }
    
}
